core/tests/protractor/accessibility.js

```diff
@@ -29,10 +29,14 @@ describe('screenreader and keyboard user accessibility features', function() {
 
   it('should skip to the main content element', function() {
     var mainContent = element(by.css('.protractor-test-main-content'));
+    var skipToMainContentLink = element(by.css('.protractor-test-skip-link'));
+    var until = protractor.ExpectedConditions;
     libraryPage.get();
     browser.actions().sendKeys(protractor.Key.TAB).perform();
-    general.waitForSystem();
-    element(by.css('.protractor-test-skip-link')).click();
+    browser.wait(
+      until.elementToBeClickable(skipToMainContentLink),
+      5000, 'Skip to main content link is not clickable');
+    skipToMainContentLink.click();
```

Comment:
browser.wait(ExpectedCondition) returns a [promise](http://www.protractortest.org/#/api?view=webdriver.WebDriver.prototype.wait) object of a [boolean](http://www.protractortest.org/#/api?view=ProtractorExpectedConditions) variable. 
With that context, I think it is better practice to process the promise and handle the success case, i.e.
```javascript
 browser.wait(
  until.elementToBeClickable(skipToMainContentLink),
  5000, 'Skip to main content link is not clickable')
  .then (function(isClickable) {
    if (isClickable) {
      // Actions!
    }
  });
```
Yes, this could lead you down to a long chain of `.then(function(isCondition))` statement. What do you think?
one

Reply:
Hi @hoangviet1993, I think we could do away with `.then` if it is not necessary. None of the examples [here](http://www.protractortest.org/#/api?view=ProtractorExpectedConditions) do so...

Reply:
Point taken! I just want a crystal clear condition set before my clicking actions due to flakiness but I guess that is just me :+1: